File: aulas/gitflow/exemplos/analise_vendas.py
"""
Exemplo prático: Análise de Vendas em Python
Este arquivo será usado para demonstrar fluxo Git
"""

import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

class AnalisadorVendas:
    """Classe para demonstrar desenvolvimento colaborativo com Git"""
    
    def __init__(self, dados_vendas):
        self.dados = dados_vendas
        print("Analisador de vendas inicializado")
    
    def resumo_basico(self):
        """Análise básica dos dados"""
        # TODO: Na aula, cada aluno criará uma branch para 
        # implementar funcionalidades específicas
        
        print("=== Resumo Básico ===")
        print(f"Total de produtos: {len(self.dados)}")
        print(f"Vendas totais: {self.dados['vendas'].sum()}")
        return self.dados.describe()
    
    def criar_visualizacao(self):
        """TODO: Implementar na branch feature/visualizacao-python"""
        # Esta função será desenvolvida por um dos alunos
        print("Criando visualização... (TODO)")
        pass
    
    def exportar_relatorio(self, formato='html'):
        """TODO: Implementar na branch feature/export-python"""
        # Esta função será desenvolvida por outro aluno
        print(f"Exportando relatório em {formato}... (TODO)")
        pass
    
    def analise_por_regiao(self):
        """TODO: Implementar na branch feature/analise-regional"""
        # Mais uma funcionalidade para demonstrar branches
        print("Analisando vendas por região... (TODO)")
        pass

# Dados de exemplo
def criar_dados_exemplo():
    """Cria dataset de exemplo para demonstrações"""
    return pd.DataFrame({
        'produto': ['Produto A', 'Produto B', 'Produto C', 'Produto D', 'Produto E'],
        'vendas': [150, 230, 180, 120, 200],
        'regiao': ['Norte', 'Sul', 'Centro', 'Norte', 'Sul'],
        'mes': ['Jan', 'Jan', 'Fev', 'Fev', 'Mar']
    })

# Executar exemplo básico
if __name__ == "__main__":
    # Criar dados de teste  
    dados = criar_dados_exemplo()
    
    # Inicializar analisador
    analisador = AnalisadorVendas(dados)
    
    # Executar análise básica
    resultado = analisador.resumo_basico()
    print(resultado)

File: aulas/gitflow/exemplos/analise_vendas.R
# Exemplo prático: Análise de Vendas - Alteração feita por Oscar a partir da branch 'desenvolvimento'
# Nova alteração a partir do desenvolvimento.
# Este arquivo será usado para demonstrar fluxo Git

# Carregar bibliotecas necessárias
library(ggplot2)
library(dplyr)

# Função principal de análise
analise_vendas <- function(dados_vendas) {
  # TODO: Na aula, cada aluno criará uma branch para implementar 
  # uma funcionalidade específica nesta função
  
  cat("Iniciando análise de vendas...\n")
  
  # Base para expansion
  resultado <- dados_vendas %>%
    summary()
    
  return(resultado)
}

# Função de exemplo para branch feature/visualizacao
criar_grafico_vendas <- function(dados) {
  # TODO: Implementar na branch feature/visualizacao
  # Esta função será desenvolvida por um dos alunos
  
  cat("Criando gráfico de vendas...\n")
  # Placeholder para implementação
}

# Função de exemplo para branch feature/relatorio
gerar_relatorio <- function(dados) {
  # TODO: Implementar na branch feature/relatorio  
  # Esta função será desenvolvida por outro aluno
  
  cat("Gerando relatório de vendas...\n")
  # Placeholder para implementação
}

# Dados de exemplo para teste
dados_exemplo <- data.frame(
  produto = c("A", "B", "C", "D", "E"),
  vendas = c(150, 230, 180, 120, 200),
  regiao = c("Norte", "Sul", "Centro", "Norte", "Sul")
)

# Executar análise básica
resultado <- analise_vendas(dados_exemplo)
print(resultado)

# Feature - André

funcao_andre <- function(...){print(paste0("Printa o seguinte termo: ", ...))}

funcao_andre("aaa")
